docs/contributor/arch/016-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,118 @@
+The converter functions support a broad variety of data types, including xml, json, and string; not all of these functions are required for the current use cases, basic operation like string comparison, formatting and data type checks could be sufficient. 
+
+The OTTL supports three types of error modes to handle errors in the OTTL expression execution:
+- `ignore` : The processor ignores errors returned by conditions, logs them, and continues on to the next condition.
+- `slient` : The processor ignores errors returned by conditions, does not log them, and continues on to the next condition.
+- `propagate` : The processor returns the error up the pipeline. This will result in the payload being dropped from the collector.
+
+The recommended error mode is `ignore` and this will be used as default configuration.
+
+The OTTL context will be embedded in the OTTL statements, [this is still in progress ](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29017) and will be available in the upcoming beta version. The solution will not implement the context as configuration parameter.
+
+To ensure data consistency and sampling efficiency, the custom OTTL transformation and filtering processor will be end of the pipeline chain before exporters.
+
+The OTel configuration :
+
+```yaml
+service:
+    pipelines:
+        metrics/test-output:
+            receivers:
+                - routing/test
+                - forward/test
+            processors:
+                - filter/drop-if-input-source-runtime
+                - filter/drop-if-input-source-prometheus
+                - filter/drop-if-input-source-istio
+                - transform/set-instrumentation-scope-kyma
+                - resource/insert-cluster-name
+                - resource/delete-skip-enrichment-attribute
+                - transform/custom-transform-processor
+                - filter/custom-filter-processor
+                - batch
+            exporters:
+                - otlp/test
+```
+
+### Risk
+
+The OTTL library is still in its alpha phase, meaning that updates to the package may introduce breaking changes, including syntax and semantic updates or bugs. This poses a risk of disrupting running pipelines.
+
+### Mitigation
+
+To mitigate this risk, we suggest the following measures:
+- The [beta version](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28892) of the OTTL library will be used in the OTel Collector to ensure that the library is stable and reliable.
+- New Versions with Breaking Changes: New versions with [breaking changes](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#breaking-changes) of the OTTL library will be rolled out with backward compatibility. This will help users adapt changes and reduce the chance of widespread disruption.
```

Comment:
is this a measure _we_ can take? Do we make sure that new versions are rolled out with backward compatibility? The passive phrase "will be rolled out" doesn't indicate who takes care of that; or what our contribution to this mitigative action is.

Reply:
The point reworked